### PR TITLE
sw-6545 adding priority for managers #minor

### PR DESCRIPTION
**Pull request by Nicholassully**

### CODE REVIEW

**Review by denialanderror — CHANGES_REQUESTED** [review]

File: json-server/config/success-rerouter.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ module.exports = (req, res, next) => {
 
         if (successRoute) {
             req.method = "GET";
-            req.url = `/successes/${successRoute}`;
+            req.url = `/successes-${successRoute}`;
```
[the remark this answers is not in the document]
Comment (Nicholassully) [reply json-server/config/success-rerouter.js:12]: I changed it as Go was complaining about the Id being a string and not being able to unmarshal it to a int. Thats way I changed it

File: web/template/layout/table.gotmpl [hunk 2]
```diff
@@ -32,7 +32,11 @@
     <tbody class="govuk-table__body">
       {{if ne .TaskList.TotalTasks 0}}
         {{ range .TaskList.WholeTaskList }}
-          <tr class="govuk-table__row">
+            {{ if .ApiPriorityTask }}
+            <tr class="govuk-table__row priority">
+            {{ else }}
+            <tr class="govuk-table__row">
+            {{ end }}
```
[the remark this answers is not in the document]
Comment (Nicholassully) [reply web/template/layout/table.gotmpl:39]: I am unsure how to do that in class but happy to talk about it.

File: internal/server/workflow.go [hunk 3]
```diff
@@ -278,3 +271,20 @@ func loggingInfoForWorkflow(client WorkflowInformation, tmpl Template, defaultWo
 		return tmpl.ExecuteTemplate(w, "page", vars)
 	}
 }
+
+func successMessageForReassignAndPrioritiesTasks(vars WorkflowVars, assignTeam string, prioritySelected string, selectedTasks []string, assigneeDisplayName string) string {
+	errorCheck := len(vars.Errors) == 0
```
[the remark this answers is not in the document]
Comment (Nicholassully) [reply internal/server/workflow.go:276]: done

File: web/template/layout/table.gotmpl (hunk 2 above)
[the remark this answers is not in the document]
Comment (denialanderror) [reply web/template/layout/table.gotmpl:39]: Sorry, Github's formatting broke and ignored the code I wrote. See updated comment.

File: json-server/config/success-rerouter.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (denialanderror) [reply json-server/config/success-rerouter.js:12]: Go shouldn't care and the last part of a URL doesn't have to be an ID of any sort. This works fine in the Hubs so I'd suggest checking how it works there if it's not working here. Let me know if you need a hand.

File: web/template/layout/table.gotmpl (hunk 2 above)
[the remark this answers is not in the document]
Comment (Nicholassully) [reply web/template/layout/table.gotmpl:39]: ah cool thank you

File: json-server/config/success-rerouter.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Nicholassully) [reply json-server/config/success-rerouter.js:12]: I think that we will need a chat than as I really tried many different ways :/

File: internal/server/workflow.go [hunk 4]
```diff
@@ -151,16 +145,16 @@ func loggingInfoForWorkflow(client WorkflowInformation, tmpl Template, defaultWo
...
 
+			prioritySelected := r.FormValue("priority")
 			// Attempt to save
-			err = client.AssignTasksToCaseManager(ctx, newAssigneeIdForTask, selectedTasks)
+			assigneeDisplayName, err := client.AssignTasksToCaseManager(ctx, newAssigneeIdForTask, selectedTasks, prioritySelected)
+
 			if err != nil {
 				logger.Print("AssignTasksToCaseManager: " + err.Error())
-				return err
+				return errors.New("Only managers can set priority on tasks")
```
[the remark this answers is not in the document]
Comment (Nicholassully) [reply internal/server/workflow.go]: done

**Review by denialanderror — APPROVED** [review]

**Pull request merged**